taichi/runtime/llvm/runtime.cpp

```diff
@@ -1550,6 +1550,16 @@ void stack_push(Ptr stack, size_t max_num_elements, std::size_t element_size) {
 }
 
 #include "internal_functions.h"
+
+void set_partial_bits_b32(u32*ptr, u32 offset, u32 bits, u32 value) {
+  u32 mask = ~(u32)0;
+  mask = ((mask << offset) >> (32 - offset - bits));
+  u32 new_value = 0;
+  do {
+    u32 old_value = *ptr;
+    new_value = (old_value & (~mask)) | (value << offset);
+  } while (!atomic_exchange_u32(ptr, new_value));
```

Comment:
I believe here we should use `__atomic_compare_exchange` instead of `atomic_exchange_u32`. See

https://github.com/taichi-dev/taichi/blob/dea88d0531e0097a355b024753c2850ca7622d56/taichi/runtime/llvm/atomic.h#L63-L86